### PR TITLE
feat(tax): normalize country and province code

### PR DESCRIPTION
**Pull request by srindom**
**What**
- Ensures that country and province codes are always stored as lower case.
- Ensures that calculation context normalizes input before getting region rates.

### CODE REVIEW

File: packages/tax/src/services/tax-module-service.ts [hunk 1]
```diff
@@ -246,12 +227,20 @@ export default class TaxModuleService<
       {
         $or: [
           {
-            country_code: calculationContext.address.country_code,
+            country_code: this.normalizeRegionCodes(
```
Comment (sradevski) [inline packages/tax/src/services/tax-module-service.ts]: A nicer approach is to have a single `normalizeTaxLinesCalculationContext` function that will normalize all input at the beginning of the method, and then you don't need to worry about forgetting to normalize in one place anymore.

File: packages/tax/src/services/tax-module-service.ts [hunk 2]
```diff
@@ -278,6 +267,38 @@ export default class TaxModuleService<
     return toReturn.flat()
   }
 
+  private prepareTaxRegionInputForCreate(
+    data: TaxTypes.CreateTaxRegionDTO | TaxTypes.CreateTaxRegionDTO[]
+  ) {
+    const input = Array.isArray(data) ? data : [data]
+    return input.reduce(
```
Comment (sradevski) [inline packages/tax/src/services/tax-module-service.ts]: This code is really difficult to read and understand. Maybe we can use a simple loop instead (usually long `reduce` functions make code less readable), and try to structure the logic a bit better?

File: packages/tax/src/services/tax-module-service.ts (hunk 1 above)
Comment (srindom) [reply packages/tax/src/services/tax-module-service.ts]: Great push! Thanks

File: packages/tax/src/services/tax-module-service.ts (hunk 2 above)
Comment (srindom) [reply packages/tax/src/services/tax-module-service.ts]: Gave it another shot - wdyt?

**Review by olivermrbl — APPROVED** [review]
LGTM

**Review by sradevski — APPROVED** [review]
Much better, thanks!

**Pull request merged**